app/scripts/metamask-controller.js

```diff
@@ -4165,9 +4165,42 @@ export default class MetamaskController extends EventEmitter {
   }
 
   async addTransaction(txParams, options) {
+    ///: BEGIN:ONLY_INCLUDE_IF(blockaid)
+    let securityAlertResponse;
+
+    try {
+      const request = {
+        method: 'eth_sendTransaction',
+        id: options.actionId,
+        origin: options.origin,
+        params: [
+          {
+            from: txParams.from,
+            to: txParams.to,
+            value: txParams?.value,
+            data: txParams?.data,
```

Comment:
txParams will always have `from` and 'to` fields, but value and data are optional.

Reply:
@segun optional chaining optional chaining is used to short circuit when the object or function being accessed are null or undefined (i.e. if txParams is undefined, then it will short circuit txParams.value and resolve to undefined instead of throwing an error).
So @legobeat point was:
1. either txParams is required to be an object, and thus it does not make sense to use optional chaining (we can just `txParams.value` and `txParams.data`)
2. or txParams is indeed optional and in that case we need to use optional chaining to access the `from` and `to` properties as well.

Reply:
Oh I see.